core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -628,12 +565,12 @@ public void onConnectionClosedByServer(Status status) {
 
                 @Override
                 public void onInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, true);
+                  isIdle = false;
                 }
 
                 @Override
                 public void onNotInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, false);
+                  isIdle = true;
```

Comment:
I wouldn't try to remove `inUseStateAggregator`. I don't see much gain to be had there. Instead, I'd focus on making `exitIdleMode` able to avoid the lock in heavy-use scenarios, i.e., when it doesn't need to do anything because the channel is already in-use.